src/main/java/net/minestom/server/particle/data/DustColorTransitionParticleData.java

```diff
@@ -0,0 +1,52 @@
+package net.minestom.server.particle.data;
+
+import net.minestom.server.network.NetworkBuffer;
+import org.jetbrains.annotations.NotNull;
+
+public record DustColorTransitionParticleData(float fromRed, float fromGreen, float fromBlue, float scale, float toRed, float toGreen, float toBlue) implements ParticleData {
```

Comment:
What about instead making these Adventure RGBLike or whatever the type is? Or maybe an overload?